src/fragmentarium/domain/Genres.ts

```diff
@@ -0,0 +1,90 @@
+import produce, { castDraft, Draft, immerable } from 'immer'
+import _ from 'lodash'
+
+export class Genre {
+  [immerable] = true
+  readonly category: ReadonlyArray<string>
+  readonly uncertain: boolean
+
+  constructor(category: string[], uncertain: boolean) {
+    this.category = category
+    this.uncertain = uncertain
+  }
+
+  setUncertain(uncertain: boolean): Genre {
+    return produce(this, (draft: Draft<Genre>) => {
+      draft.uncertain = uncertain
+    })
+  }
+
+  get toString(): string {
+    return this.category.join('-').replace(' ', '_')
+  }
+}
+
+export class Genres {
+  [immerable] = true
+  readonly genres: ReadonlyArray<Genre>
+
+  constructor(genres: Genre[]) {
+    this.genres = genres
+  }
+
+  static fromJSON(
+    genreJSON: { category: string[]; uncertain: boolean }[]
+  ): Genres {
+    return new Genres(
+      genreJSON.map(({ category, uncertain }) => new Genre(category, uncertain))
+    )
+  }
+
+  isPresent(genre: Genre): boolean {
+    return this.genres.some(
+      (element) =>
+        JSON.stringify(element.category) === JSON.stringify(genre.category)
+    )
+  }
+
+  find(genre: Genre) {
+    return _.find(
+      this.genres,
+      (elem) => JSON.stringify(elem.category) === JSON.stringify(genre.category)
```

Comment:
You might use [_.isEqual](https://lodash.com/docs/4.17.15#isEqual).

Reply:
i use `_.isEqual instead the object method `isEqualTo`. But in this case `find` will match if the `category`are equal disregarding the `uncertain` property so `_.isEqual`wouldn't work.